src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseGradingService.java

```diff
@@ -374,15 +387,38 @@ private void createFeedbackForNotExecutedTests(Result result, Set<ProgrammingExe
         result.addFeedbacks(feedbacksForNotExecutedTestCases);
     }
 
+    private boolean createFeedbackForDuplicateTests(Result result, ProgrammingExercise programmingExercise) {
+        Set<String> uniqueFeedbackNames = new HashSet<>();
+        // Find duplicate test cases from feedback which is automatic feedback
+        // Set.add() returns false if the element was already in the set, this is how we find all duplicates
```

Comment:
Maybe move this 2 lines down :)

Reply:
done in 20cd2a8efb4c9630ecd1bfc55392b3ca37e9076c